pennylane/templates/subroutines/permute.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``permute`` template.
+"""
+
+import pennylane as qml
+
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+from pennylane.wires import Wires
+
+
+@template
+def Permute(permutation, wires):
+    r"""Permutes a set of wires into a new order.
+
+    Args:
+        permutation (list): A list of wire labels that represents the new ordering of wires
+            after the permutation. The list may consist of integers or strings, so long as
+            they match the labels of the wires.
+        wires (Iterable or Wires): Wires that the template acts on. Accepts an iterable of numbers
+            or strings, or a Wires object.
+
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
```

Comment:
Could we include a small, minimal 'this is how it is used' example before the `.. UsageDetails::` directive? Since Usage Details is hidden by default, I find it is best for using it more like 'more example' or 'more advanced details'. (I think a lot of the other templates also don't include a top-level minimal example just showing usage)